### ISSUE TITLE
Version 2.3.0 incompatible with SpringBoot - KmsMasterKey class does not load

### ISSUE THREAD

**Issue by sparecycles** [aws/aws-encryption-sdk-java#261]
### Problem:

VersionInfo#loadUserAgent tries to load a version from a file, this fails catastrophically when this library is packed with the spring-boot-maven-plugin: the result of `ClassLoader.getSystemResourceAsStream("project.properties")` is null, triggering a NullPointerException which is not handled, which results in the following error for me.
```
java.lang.NoClassDefFoundError: Could not initialize class com.amazonaws.encryptionsdk.kms.KmsMasterKey
```

https://github.com/aws/aws-encryption-sdk-java/blob/163d79fbb43f06c09308fec295050a3fad01c751/src/main/java/com/amazonaws/encryptionsdk/kms/KmsMasterKey.java#L54-L55

https://github.com/aws/aws-encryption-sdk-java/blob/163d79fbb43f06c09308fec295050a3fad01c751/src/main/java/com/amazonaws/encryptionsdk/internal/VersionInfo.java#L27-L35

Works fine directly (IntelliJ run), but debugging the app within my docker-build clearly shows the issue:

<img width="1097" alt="is-null" src="https://user-images.githubusercontent.com/719818/122485238-46ca5100-cf8b-11eb-8c02-fbde4e96dce6.png">

<img width="1109" alt="throws-npe" src="https://user-images.githubusercontent.com/719818/122485239-47fb7e00-cf8b-11eb-8985-d832961d8ba5.png">

application.jar packed with

```xml
      <plugin>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-maven-plugin</artifactId>
        <version>2.4.2</version> <!-- EDIT: also tested latest = 2.5.1 -->
      </plugin>
```

### Solution(s):

- Handle a null resource stream, and/or catch NullPointerExceptions here as well as IOExceptions.
- Or simply hard-code the version directly in this file rather than delegating to a properties resource?
- Use a classLoader from the same jar, e.g., the following works in my debugger:

```java
properties.load(VersionInfo.class.getClassLoader().getResourceAsStream("project.properties"));
```

**Comment by dankirkd** [comment]
Any idea when this fix will be released?

**Comment by alex-chew** [comment]
Sorry for the delay - the fix is now released as of v2.3.1: https://github.com/aws/aws-encryption-sdk-java/releases/tag/v2.3.1

**Comment by sparecycles** [comment]
Fix confirmed working in deployment.  Thanks @alex-chew !
